riyaz/db.py

```diff
@@ -289,6 +308,50 @@ def set(cls, key, value):
         return kv.save()
 
 
+class Asset(Document):
+    _TABLE = "asset"
+
+    collection: str
+    collection_id: int
+    filename: str
+
+    filesize: Optional[int]
+    created: Optional[datetime]
+    last_modified: Optional[datetime]
+
+    def _get_full_identifier(self):
+        return f"{self.collection}/{self.collection_id}/{self.filename}"
+
+    def _construct_asset_path(self):
+        assert "/" not in self.filename, "filename should not be a path"
+
+        full_identifier = self._get_full_identifier()
+        return Path(config.assets_path) / full_identifier
+
+    def get_url(self):
+        assets_root = "/assets/"
+
+        full_identifier = self._get_full_identifier()
+        return f"{assets_root}{full_identifier}"
+
+    def save_file(self, on_disk_path: Path):
+        assert on_disk_path.is_file()
+
+        asset_path = self._construct_asset_path()
+
+        stat = on_disk_path.stat()
+        filesize, created, last_modified = (
+            stat.st_size, int(stat.st_ctime), int(stat.st_mtime))
```

Comment:
Shouldn't you take the creation time and last modified time when the asset is created and updated in the server rather than looking at the file? 

If I upload an asset with same name twice, should its creation time stay same as the first time or change?

Reply:
Oh okay, I thought we were looking for file metadata.

> If I upload an asset with same name twice, should its creation time stay same as the first time or change?

Shouldn't it change, because it's a new asset?

But, if not file metadata, are we saving this for timestamping rows?
In that case, I think we can give that responsibility to `Document.save` to add created and last_modified.

What do you think?

Reply:
Done.
Added an `update_timestamps()` method that will take `datetime.now()` and set created (conditionally) and last_modified to now.